server/api/order.js

```diff
@@ -18,40 +68,106 @@ router.get("/:orderId", async (req, res, next) => {
     const order = await Order.findByPk(req.params.orderId, {
       include: OrderDetails,
     });
+    console.log("order:", order);
+
     res.json(order);
   } catch (err) {
     next(err);
   }
 });
 
-// router.post("/", async (req, res, next) => {
-//   try {
-//     res.json(
-//       await Order.create(req.body, {
-//         include: OrderDetails,
-//       })
-//     );
-//   } catch (err) {
-//     next(err);
-//   }
-// });
+router.get("/:orderId/:orderDetailsId", async (req, res, next) => {
+  try {
+    const order = await OrderDetails.findByPk(req.params.orderDetailsId);
+    res.json(order);
+  } catch (err) {
+    next(err);
+  }
+});
 
-router.post("/", async (req, res, next) => {
+router.post("/user/:userId/:productId", async (req, res, next) => {
   try {
-    console.log("//////////", req.body);
-    const { orderDetailsParamsList, ...orderParams } = req.body;
-    const order = await Order.create(orderParams);
-    for (let i = 0; i < orderDetailsParamsList.length; i++) {
-      const params = orderDetailsParamsList[i];
-      const orderDetails = await OrderDetails.create(params);
-      await orderDetails.setOrder(order);
+    //grab a user's unfulfilled cart/order
+    const existingCart = await Order.findOne({
```

Comment:
i think this one should be `findOrCreate` as well, regardless of how it is used in the front end you want it to work